File: LeetCode Algorithms/350.两个数组的交集-ii.py
#
# @lc app=leetcode.cn id=350 lang=python3
#
# [350] 两个数组的交集 II
#

# @lc code=start
class Solution:
    @classmethod
    def intersect(self, nums1, nums2):
        res = []
        _index_1, _index_2 = 0, 0
        nums1 = sorted(nums1)
        nums2 = sorted(nums2)
        
        while True:
            if _index_1 == len(nums1) or _index_2 == len(nums2):
                break
            if nums1[_index_1] == nums2[_index_2]:
                res.append(nums1[_index_1])
                _index_1 += 1
                _index_2 += 1
            elif nums1[_index_1] > nums2[_index_2]:
                _index_2 += 1
            else: _index_1 += 1

        return res

# @lc code=end

print (Solution.intersect([1,2,2,1], [2,2]))

File: LeetCode Algorithms/350.两个数组的交集-ii.cpp
/*
 * @lc app=leetcode.cn id=350 lang=cpp
 *
 * [350] 两个数组的交集 II
 */

// @lc code=start
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        vector<int> res;
        // 方法一：STL 求交函数
        // set_intersection(nums1.begin(), nums1.end(), nums2.begin(), nums2.end(), inserter(res, res.begin()));

        // 方法二： 双指针

        
        return res;
    }
};
// @lc code=end

